### PR TITLE
Fix try builds of the system add-on

### PR DESCRIPTION
**Pull request by Osmose**
There are failures with our add-on that can happen only during
packaging of Firefox, such as duplicate bootstrap.js files. We should
run that step in TaskCluster CI so we can catch those issues early.

This should catch the issue from #718 and fail. I'm still working on a fix, but want to run this in the meantime.

### CODE REVIEW

File: recipe-client-addon/bin/tc/package.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+curl --location --fail --silent --show-error "$BUILD_RESULT" | tar xz
+
+echo 'Setting up environment'
+pushd gecko-dev-master
+source /root/.cargo/env
+python2.7 ./python/mozboot/bin/bootstrap.py --no-interactive --application-choice=browser
+source /root/.cargo/env
+
+echo 'Packaging Firefox'
+./mach package
```
Comment (mythmon) [inline recipe-client-addon/bin/tc/package.sh]: Might as well put the results of this in an artifact?

Comment (Osmose) [inline recipe-client-addon/bin/tc/package.sh]: Sure, if we can find a good way to determine the name of that directory. I wouldn't consider it a blocker though.

**Comment by Osmose** [comment]
Okay, this is ready. The diff looks big, but most of the changes are just copy-pastes from bootstrap.js to ShieldRecipeClient.jsm.

File: recipe-client-addon/bootstrap.js [hunk 2]
```diff
@@ -4,95 +4,27 @@
...
-  CleanupManager.cleanup();
-
-  if (reason === REASONS.ADDON_DISABLE || reason === REASONS.ADDON_UNINSTALL) {
-    Services.prefs.setBoolPref(PREF_SELF_SUPPORT_ENABLED, true);
-  }
+  ShieldRecipeClient.shutdown(reason);
 
+  // Unload add-on modules. We don't do this in ShieldRecipeClient so that
+  // modules are not unloaded accidentally during tests.
+  const log = LogManager.getLogger("bootstrap");
```
Comment (gijsk) [inline recipe-client-addon/bootstrap.js]: We may or may not have called LogManager.configure() here, per https://bugzilla.mozilla.org/show_bug.cgi?id=1351465#c5 . I don't know if that should be a serious concern, but I wanted to flag this up.

**Comment by Osmose** [comment]
Updated. I moved the logging setup to before the early return. There's some opportunity for refactoring this to be better but I'm punting on that for now.

**Pull request merged**